File: src/main/java/yt/szczurek/hyrmur/bedwars/command/BedwarsMapListCommand.java
package yt.szczurek.hyrmur.bedwars.command;

import com.hypixel.hytale.builtin.instances.InstancesPlugin;
import com.hypixel.hytale.server.core.Message;
import com.hypixel.hytale.server.core.command.system.CommandContext;
import com.hypixel.hytale.server.core.command.system.basecommands.AbstractAsyncCommand;
import com.hypixel.hytale.server.core.util.message.MessageFormat;

import javax.annotation.Nonnull;
import java.util.List;
import java.util.concurrent.CompletableFuture;

public class BedwarsMapListCommand extends AbstractAsyncCommand {
    public BedwarsMapListCommand() {
        super("list", "server.commands.bedwars.map.list.desc");
    }

    @Nonnull
    @Override
    public CompletableFuture<Void> executeAsync(@Nonnull CommandContext context) {
//        List<String> instanceAssets = InstancesPlugin.get().getInstanceAssets();
//        context.sendMessage(
//                MessageFormat.list(Message.translation("server.commands.instances.edit.list.header"), instanceAssets.stream().map(Message::raw).toList())
//        );
        return CompletableFuture.completedFuture(null);
    }
}

File: src/main/kotlin/yt/szczurek/hyrmur/bedwars/command/BedwarsMapListCommand.kt
package yt.szczurek.hyrmur.bedwars.command

import com.hypixel.hytale.server.core.Message
import com.hypixel.hytale.server.core.command.system.CommandContext
import com.hypixel.hytale.server.core.command.system.basecommands.AbstractAsyncCommand
import com.hypixel.hytale.server.core.util.message.MessageFormat
import yt.szczurek.hyrmur.bedwars.asset.BedwarsMap
import java.util.concurrent.CompletableFuture

class BedwarsMapListCommand : AbstractAsyncCommand("list", "server.commands.bedwars.map.list.desc") {

    override fun executeAsync(ctx: CommandContext): CompletableFuture<Void> {
        val mapAssets = BedwarsMap.assetMap.assetMap.keys
        ctx.sendMessage(
            MessageFormat.list(HEADER, mapAssets.stream().map(Message::raw).toList())
        )
        return CompletableFuture.completedFuture(null)
    }

    companion object {
        val HEADER = Message.translation("server.commands.bedwars.map.list.header")
    }
}
